src/CharityVaultFactory.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import {ERC20} from "solmate/erc20/ERC20.sol";
+import {Bytes32AddressLib} from "./libraries/Bytes32AddressLib.sol";
+import "./CharityVault.sol";
+
+/// @title Fuse Charity Vault Factory
+/// @author Transmissions11, JetJadeja, Andreas Bigger
+/// @notice Charity wrapper for vaults/VaultFactory.
+contract CharityVaultFactory {
+    using Bytes32AddressLib for *;
+
+    /*///////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Emitted when `deployCharityVault` is called.
+    /// @param underlying The underlying token used in the vault.
+    /// @param vault The new charity vault deployed that accepts the underlying token.
+    event CharityVaultDeployed(ERC20 underlying, CharityVault vault);
+
+    /*///////////////////////////////////////////////////////////////
+                           STATEFUL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Deploy a new CharityVault contract that supports a specific underlying asset.
+    /// @dev This will revert if a vault with the token has already been created.
+    /// @param underlying Address of the ERC20 token that the Vault will earn yield on.
+    /// @param charity donation address
+    /// @param feePercent percent of earned interest sent to the charity as a donation
+    /// @return cvault The newly deployed CharityVault contract.
+    function deployCharityVault(ERC20 underlying, address payable charity, uint256 feePercent) external returns (CharityVault cvault) {
+        // Compute a CharityVault Salt
+        bytes32 cvaultSalt = keccak256(
+            abi.encodePacked(
+                address(underlying),
+                charity,
+                feePercent
+            )
+        );
```

Comment:
why not just inline this for the salt arg?

Reply:
inline?

Reply:
like remove this variable and replace uses of `cvaultSalt` with 

```
keccak256(
            abi.encodePacked(
                address(underlying),
                charity,
                feePercent
            )
        )
``` 

that way u dont waste gas loading it into memory and then reading from it only once